projects/extension/src/background/types.ts

```diff
@@ -3,6 +3,17 @@ import { AppMediator } from './AppMediator';
 import EventEmitter from 'eventemitter3';
 import StrictEventEmitter from 'strict-event-emitter-types';
 
+export interface HealthResponse {
+  isSyncing: boolean;
+  peers: number;
+  shouldHavePeers: boolean;
+}
+
+export interface ChainInstance {
+  chain: smoldot.SmoldotChain
+  healthChecker: smoldot.HealthChecker
```

Comment:
Similarly, this exposes an implementation detail, I don't see why you would ever need this field.

Reply:
This is needed from both mocks and ConnectionManager. I can though duplicated it in relevant files.

Reply:
This should be removed as well, consequently. There's no need for anything outside `AppMediator` to know about the health checker.

Reply:
This is already removed